### PR TITLE
chore: Upgrade packages + Migrate to Boostrap 5

### PR DESCRIPTION
**Pull request by michael-wisely-gravwell**
This PR doesn't address any particular issue, but it's related to https://github.com/gravwell/wiki/issues/838

This PR proposes bumping the version of nixpkgs that we're pulling, effectively upgrading all relevant packages.

As a result of the pkgs bump, we're using pydata-sphinx-theme@0.15.1 which uses Bootstrap 5. Boostrap 5 has some backward-incompatible changes, so this PR also proposes updating relevant source files to use Bootstrap 5 stuff.

Unfortunately, pydata-sphinx-theme@0.15.1 doesn't appear to make the `bootstrap` global available. Out of the box, one can't [enable Popovers in the way the Boostrap docs recommend](https://getbootstrap.com/docs/5.3/components/popovers/#enable-popovers). I [forked the theme](https://github.com/michael-wisely-gravwell/pydata-sphinx-theme/tree/globally-available-bootstrap), added a line to the bundled bootstrap.js file, built the theme into a Python wheel, and included it in this repo. By vendorizing the lib, I was able to pull in the change to make `bootstrap` globally available.

In summary, this PR proposes...

- upgrading docs-building packages
- vendorizing pydata-sphinx-theme
- updating docs and supporting files to use Boostrap 5

### CODE REVIEW

**Review by ashnwade — DISMISSED** [review]
It looks like this is no longer using our custom Not Found page (`404.md`). Can we fix that?

**Review by ashnwade — APPROVED** [review]
